web-module/src/main/webapp/resources/structEditJS/tinyMCE4/loadTinyMCE-async.js

```diff
@@ -96,42 +96,63 @@
 
   var lazyLoadTinyMCEforTab = function(event) {
     try {
-      var tabBodyId = event.memo.newTabId;
-      var tinyMceAreas = $(tabBodyId).select('textarea.mceEditor');
-      console.log('lazyLoadTinyMCEforTab: for tabBodyId ', tabBodyId, tinyMceAreas);
-      tinyMceAreas.each(function(editorArea) {
-        tinyMCE.execCommand("mceAddEditor", false, editorArea.id);
-      });
+      var tabBodyId = event.memo.newTabBodyId;
+      if (tinyConfigLoaded) {
+        var tinyMceAreas = $(tabBodyId).select('textarea.mceEditor');
+        console.log('lazyLoadTinyMCEforTab: for tabBodyId ', tabBodyId, tinyMceAreas);
+        tinyMceAreas.each(function(editorArea) {
+          tinyMCE.execCommand("mceAddEditor", false, editorArea.id);
+        });
+        console.log('lazyLoadTinyMCEforTab: finish', tabBodyId);
+      } else {
+        console.log('lazyLoadTinyMCEforTab: skip mceAddEditor because tinyConfig not yet loaded.',
+            tabBodyId);
+      }
     } catch (exp) {
       console.error("lazyLoadTinyMCEforTab failed. ", exp);
     }
   };
 
+  var getUninitializedMceEditors = function(mceParentElem) {
+    console.log('getUninitializedMceEditors: start ', mceParentElem);
+    var mceEditorsToInit = new Array();
+    $(mceParentElem).select('textarea.mceEditor').each(function(editorArea) {
+      var notInitialized = !tinymce.get(editorArea.id);
+      console.log('getUninitializedMceEditors: found new editorArea ', editorArea.id,
+          notInitialized);
+      if (notInitialized) {
+        mceEditorsToInit.push(editorArea.id);
+      }
+    });
+    console.log('getUninitializedMceEditors: returns ', mceParentElem, mceEditorsToInit);
+    return mceEditorsToInit;
+  };
+
   var delayedEditorOpeningHandler = function(event) {
-    console.log('delayedEditorOpeningHandler: start');
-    var mceEditorAreaAvailable = ($$('#tabMenuPanel .mceEditor').size() > 0);
-    if (!finishedCelRTE_tinyMCE_Load && mceEditorAreaAvailable) {
+    console.log('delayedEditorOpeningHandler: start ', event.memo);
+    var mceParentElem = event.memo.tabBodyId || "tabMenuPanel";
+    var mceEditorAreaAvailable = (getUninitializedMceEditors(mceParentElem).size() > 0);
+    if (mceEditorAreaAvailable) {
```

Comment:
consider refactoring `getUninitializedMceEditors` to `hasUninitializedMceEditors`, since it's the only use case

Reply:
now reusing it in lazyLoadTinyMCEforTab :-)